### ISSUE TITLE
Javascript error:"Cannot read property 'wikiScript' of undefined" (MW 1.29+)

### ISSUE THREAD

**Issue by Nikerabbit** [SemanticMediaWiki/SemanticMediaWiki#2129]
### Setup and configuration

- SMW version: 2.4.4
- MW version: 1.29 master
- PHP version: HHVM 3.15.0
- DB (MySQL etc.): 5.5.50-MariaDB

### Issue
Javascript error:"Cannot read property 'wikiScript' of undefined"

```
"TypeError: Cannot read property 'wikiScript' of undefined
    at HTMLDocument.eval (eval at <anonymous> (http://88.99.39.75/w/load.php?debug=false&lang=fi&modules=jquery%2Cmediawiki&only=scripts&skin=vector&version=0v09rhw:4:681), <anonymous>:74:1160)
    at fire (http://88.99.39.75/w/load.php?debug=false&lang=fi&modules=jquery%2Cmediawiki&only=scripts&skin=vector&version=0v09rhw:45:122)
    at Object.add [as done] (http://88.99.39.75/w/load.php?debug=false&lang=fi&modules=jquery%2Cmediawiki&only=scripts&skin=vector&version=0v09rhw:45:656)
    at jQuery.fn.init.jQuery.fn.ready (http://88.99.39.75/w/load.php?debug=false&lang=fi&modules=jquery%2Cmediawiki&only=scripts&skin=vector&version=0v09rhw:49:63)
    at eval (eval at <anonymous> (http://88.99.39.75/w/load.php?debug=false&lang=fi&modules=jquery%2Cmediawiki&only=scripts&skin=vector&version=0v09rhw:4:681), <anonymous>:74:1003)
    at eval (eval at <anonymous> (http://88.99.39.75/w/load.php?debug=false&lang=fi&modules=jquery%2Cmediawiki&only=scripts&skin=vector&version=0v09rhw:4:681), <anonymous>:74:1536)
    at Object.<anonymous> (http://88.99.39.75/w/load.php?debug=false&lang=fi&modules=jquery%2Cmediawiki&only=scripts&skin=vector&version=0v09rhw:161:648)
    at fire (http://88.99.39.75/w/load.php?debug=false&lang=fi&modules=jquery%2Cmediawiki&only=scripts&skin=vector&version=0v09rhw:45:122)
    at Object.add [as done] (http://88.99.39.75/w/load.php?debug=false&lang=fi&modules=jquery%2Cmediawiki&only=scripts&skin=vector&version=0v09rhw:45:656)
    at Object.always (http://88.99.39.75/w/load.php?debug=false&lang=fi&modules=jquery%2Cmediawiki&only=scripts&skin=vector&version=0v09rhw:46:874)"
```

Steps to reproduce the observation (recommendation is to use the [sandbox](http://sandbox.semantic-mediawiki.org)):

Go to Special:SearchByProperty (e.g. http://88.99.39.75/wiki/Toiminnot:SearchByProperty)
Does not happen in debug mode.

This leads me to believe there is a missing dependency to mediawiki.util module somewhere.

**Comment by mwjames** [comment]
> "TypeError: Cannot read property 'wikiScript' of undefined

As for 1.27, I'm unable to replicate the issue and [0] (MW 1.28) doesn't show those reported symptoms therefore it is likely related to MW 1.29/MW-core.

> MW version: 1.29 master

I cannot test any release beyond MW 1.27 (MW 1.28+ broke my postgres test environment). 

> This leads me to believe there is a missing dependency to mediawiki.util module somewhere.

This wasn't an issue before MW 1.29. Did something change? What is require or expected here? [1] contains the registration for modules and its dependencies.

[0] https://sandbox.semantic-mediawiki.org/wiki/Sp%C3%A9cial:Recherche_par_propri%C3%A9t%C3%A9
[1] https://github.com/SemanticMediaWiki/SemanticMediaWiki/blob/master/res/Resources.php
[2] https://github.com/SemanticMediaWiki/SemanticMediaWiki/search?utf8=%E2%9C%93&q=wikiScript

**Comment by Nikerabbit** [comment]
MW 1.29 might be more asynchronous and more likely to trigger race conditions for missing deps.

```
./res/smw/special/ext.smw.special.property.js:          $( '#smw-property-input, .smw-property-input' ).autocomplete({
./res/smw/special/ext.smw.special.property.js-                  serviceUrl: mw.util.wikiScript( 'api' ),
./res/smw/special/ext.smw.special.property.js-                  dataType: 'json',
```

`ext.smw.property` does not have `mediawiki.util` as an explicit or implicit dependency, even though it is using it. I recommend adding it.

**Comment by mwjames** [comment]
> ext.smw.property does not have mediawiki.util as an explicit or implicit dependency, even though it is using it. I recommend adding it.

Added with #2149 and in case it is expected to have this available in 2.4.5 then a backport is required.

**Comment by kghbln** [comment]
Backport to 2.4.x now available.
